byterun/array.c

```diff
@@ -35,9 +35,12 @@ CAMLexport int caml_is_double_array(value array)
 
 CAMLprim value caml_array_get_addr(value array, value index)
 {
+  CAMLparam2(array, index);
+  CAMLlocal1(x);
   intnat idx = Long_val(index);
   if (idx < 0 || idx >= Wosize_val(array)) caml_array_bound_error();
-  return Field(array, idx);
+  caml_read_field(array, idx, &x);
```

Comment:
Do we need `CAMLParam`/`CAMLlocal` in this function at all? The variables are not live across this safe point.